lib/visitor.js

```diff
@@ -7,9 +7,23 @@ const underscoreCode = "_".charCodeAt(0);
 // https://github.com/benjamn/ast-types/blob/master/lib/path-visitor.js
 
 function Visitor() {
-  this.visit = this.visit.bind(this);
-  this.visitWithoutReset = this.visitWithoutReset.bind(this);
-  this.visitChildren = this.visitChildren.bind(this);
+  const that = this;
+  const visit = that.visit;
+  const visitWithoutReset =  that.visitWithoutReset;
+  const visitChildren = that.visitChildren;
+
+  // Avoid slower `Function#bind` in Node < 7.
+  this.visit = function () {
+    visit.apply(that, arguments);
+  };
```

Comment:
Is the arg signature of `visit` potentially infinite? or does it max out at like 3 args?